filter.go

```diff
@@ -0,0 +1,148 @@
+package authzed
+
+import (
+	"context"
+	"errors"
+	"reflect"
+
+	api "github.com/authzed/authzed-go/arrakisapi/api"
+	"github.com/authzed/authzed-go/x/parallel"
+)
+
+// NewFilterIter returns a FilterIter that filters the provided slice by access
```

Comment:
This is hard to read... maybe add "by whether each item is accessible to the provided user"?